model/src/main/java/ru/training/karaf/repo/AuthorRepoImpl.java

```diff
@@ -0,0 +1,75 @@
+package ru.training.karaf.repo;
+
+import org.apache.aries.jpa.template.JpaTemplate;
+import org.apache.aries.jpa.template.TransactionType;
+import ru.training.karaf.model.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AuthorRepoImpl implements AuthorRepo {
+
+    private JpaTemplate template;
+
+    public AuthorRepoImpl(JpaTemplate template) {
+        this.template = template;
+    }
+
+    @Override
+    public Collection<Author> getAll() {
+        return template.txExpr(em -> em.createNamedQuery(AuthorDo.GET, AuthorDo.class).getResultList()).stream().map(AuthorImpl::new).collect(Collectors.toList());
+    }
+
+    @Override
+    public void create(String name, int age, Collection<String> books) {
+        AuthorDo authorToCreate = new AuthorDo();
+        authorToCreate.setName(name);
+        authorToCreate.setAge(age);
+
+        List<BookDO> books1 = template.txExpr(em -> em.createNamedQuery(BookDO.GET_ALL_WHERE_IN_AUTHORS, BookDO.class).setParameter("books", books)
```

Comment:
It's better to be done in one transaction. Creating a new one gives us overhead which may lower performance.